src/main/resources/com/google/api/codegen/go/discovery_fragment.snip

```diff
@@ -9,101 +9,136 @@
+    //    and check the quota for your project at
+    //    https://console.developers.google.com/apis/api/{@apiName}
+    // 2. This sample uses Application Default Credentials for authentication.
+    //    If not already done, install the gcloud CLI from
+    //    https://cloud.google.com/sdk/ and run
+    //    `gcloud beta auth application-default login`
+    // 3. Install and update the Go dependencies by running `go get -u` in the
+    //    project directory.
+    
+    import (
+      "encoding/json"
+      "fmt"
+      "log"
+    
+      "golang.org/x/net/context"
+      "golang.org/x/oauth2/google"
+    
+      "google.golang.org/api/{@apiName}/{@apiVersion}"
+    )
+    
+    func main() {
+      ctx := context.Background()
+      # TODO(pongad): Every API except autoscaler has the string as API.CloudPlatformScope, find out why
+
+      // Authentication is provided by the `gcloud` tool when running locally
+      // and by built-in service accounts when running on GAE, GCE, or GKE.
+      // For more information, see
+      // https://developers.google.com/identity/protocols/application-default-credentials
+      {@clientName()}, err := google.DefaultClient(ctx{@scopeArg(method)})
+      if err != nil {
+        log.Fatal(err)
+      }
+
+      // Construct the {@apiName} service object for interacting with the API.
+      // You can browse other available API services and versions at
+      // https://github.com/google/google-api-go-client
+      {@serviceName()}, err := {@apiName}.New({@clientName()})
+      if err != nil {
+        log.Fatal(err)
+      }
+
+      {@initialize(method)}
+      @if context.isPageStreaming(method)
+        {@iterableCall(method)}
+      @else
+        {@compactCall(method)}
+      @end
+    }
   @end
-}
 @end
 
 @private scopeArg(method)
   @let scopeList = context.getAuthScopes(method)
     @if not(scopeList.isEmpty)
-    , {@context.getApi.getName}.{@scopeList.get(0)}Scope
+      , {@context.getApi.getName}.{@scopeList.get(0)}Scope
     @end
   @end
 @end
 
 @private compactCall(method)
   @if context.isResponseEmpty(method)
-    @if context.hasRequestField(method)
-      call := {@methodPath(method)}({@callArg(method)}).Context(ctx)
-      if err := call.Do(); err != nil {
-        // {@TODO()} Handle error.
-      }
-    @else
-      if err := {@methodPath(method)}({@callArg(method)}).Context(ctx).Do(); err != nil {
-        // {@TODO()} Handle error.
-      }
-    @end
+    if err := {@request(method)}.Context(ctx).Do(); err != nil {
+      log.Fatal(err)
+    }
   @else
-    # TODO(pongad): generate a better name for "resp"
-    resp, err := {@methodPath(method)}({@callArg(method)}).Context(ctx).Do()
+    {@responseName()}, err := {@request(method)}.Context(ctx).Do()
+    if err != nil {
+      log.Fatal(err)
+    }
+
+    // {@TODO()} Change code below to process the `{@responseName()}` object:
+    {@printName()}, err := json.MarshalIndent({@responseName()}, "", "  ")
```

Comment:
I'm not against this change but want Go team member to sign off on this before merging. @broady Do you have an opinion?